presto-benchmark/src/main/java/com/facebook/presto/benchmark/ArrayComparisonBenchmark.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark;
+
+import com.facebook.presto.testing.LocalQueryRunner;
+
+import static com.facebook.presto.benchmark.BenchmarkQueryRunner.createLocalQueryRunner;
+
+public abstract class ArrayComparisonBenchmark
+{
+    public static void main(String... args)
+    {
+        LocalQueryRunner localQueryRunner = createLocalQueryRunner();
+        new ArrayEqualsBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+        new ArrayLessThanBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+        new ArrayGreaterThanBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+        new ArrayNotEqualBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+        new ArrayLessThanOrEqualBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+        new ArrayGreaterThanOrEqualBenchmark(localQueryRunner).runBenchmark(new SimpleLineBenchmarkResultWriter(System.out));
+    }
+
+    public static class ArrayEqualsBenchmark
+            extends AbstractSqlBenchmark
+    {
+        public ArrayEqualsBenchmark(LocalQueryRunner localQueryRunner)
+        {
+            super(localQueryRunner, "array_equals", 25, 150, "select ARRAY [1, 2, 3] = ARRAY [1, 2, 3]");
```

Comment:
I think there's going to be too much overhead in the benchmark framework for these to be meaningful. Instead I would write this like `SELECT COUNT_IF(ARRAY [orderkey, orderkey + 1, orderkey + 2] = ARRAY[orderkey + 1, orderkey + 2, orderkey + 3]) FROM orders`

Alternately, you could use JMH to write a micro benchmark. I haven't tried that myself, but @martint has had a lot of success with it